### PR TITLE
Add API to get commits of PR

### LINKED ISSUE #10918: [Feature] [API ] Get Commits of a PullRequest
api witch only return commits of a PullRequest - same as list on `https://gitea/<user>/<repo>/pulls/<333>/commits` shows

proposal: add `GET /repos/{owner}/{repo}/pulls/{index}/commits`

ToDo:
 * look if GitHub v3 API has something similar

### PR DESCRIPTION
**Pull request by sebastian-sauer**
fixes #10918

Co-authored-by: Andrew Bezold <andrew.bezold@gmail.com>

based on the changes in PR https://github.com/go-gitea/gitea/pull/12118

### CODE REVIEW

File: models/list_options.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ func (opts *ListOptions) setEnginePagination(e Engine) Engine {
 func (opts *ListOptions) GetStartEnd() (start, end int) {
 	opts.setDefaultValues()
 	start = (opts.Page - 1) * opts.PageSize
-	end = start + opts.Page
+	end = start + opts.PageSize
```
Comment (sebastian-sauer) [inline models/list_options.go]: I think the old implementation had an issue as the end should be start + pageSize and not 
start + Page 
or did i miss anything here?

Comment (zeripath) [reply models/list_options.go]: agh!

This needs to get into 1.15 and backport to 1.14 can you break this out to another PR?

Comment (sebastian-sauer) [reply models/list_options.go]: sure.

Comment (sebastian-sauer) [reply models/list_options.go]: https://github.com/go-gitea/gitea/pull/16303

File: routers/api/v1/repo/pull.go [hunk 2]
```diff
@@ -1101,3 +1103,122 @@ func UpdatePullRequest(ctx *context.APIContext) {
...
+			return
+		}
+		i++
+	}
+
+	ctx.SetLinkHeader(int(totalNumberOfCommits), listOptions.PageSize)
+
+	ctx.Header().Set("X-Page", strconv.Itoa(listOptions.Page))
+	ctx.Header().Set("X-PerPage", strconv.Itoa(listOptions.PageSize))
+	ctx.Header().Set("X-Total", strconv.FormatInt(int64(totalNumberOfCommits), 10))
```
Comment (6543) [inline routers/api/v1/repo/pull.go]:
```suggestion
	ctx.Header().Set("X-Total-Count", fmt.Sprintf("%d", totalNumberOfCommits))
```

-> https://github.com/go-gitea/gitea/issues/11114

**Review by 6543 — APPROVED** [review]

File: routers/api/v1/repo/pull.go [hunk 3]
```diff
@@ -1101,3 +1103,122 @@ func UpdatePullRequest(ctx *context.APIContext) {
...
+		i++
+	}
+
+	ctx.SetLinkHeader(int(totalNumberOfCommits), listOptions.PageSize)
+
+	ctx.Header().Set("X-Page", strconv.Itoa(listOptions.Page))
+	ctx.Header().Set("X-PerPage", strconv.Itoa(listOptions.PageSize))
+	ctx.Header().Set("X-Total-Count", fmt.Sprintf("%d", totalNumberOfCommits))
+	ctx.Header().Set("X-PageCount", strconv.Itoa(totalNumberOfPages))
+	ctx.Header().Set("X-HasMore", strconv.FormatBool(listOptions.Page < totalNumberOfPages))
```
Comment (a1012112796) [inline routers/api/v1/repo/pull.go:1222]: These message is usefull for all list api, maybe it's better to add a help function named as ``ctx.SetResultHeader(...)`` to do it.

**Review by a1012112796 — APPROVED** [review]
LGTM, and suggest move this pull request to v1.15

File: routers/api/v1/repo/pull.go (hunk 3 above)
Comment (6543) [reply routers/api/v1/repo/pull.go:1222]: yes as #11114 proposed some general "SetListResultHeaders() or so

**Comment by 6543** [comment]
@sebastian-sauer thanks for contrib ;)

**Pull request merged**